test/tstDistributedTree.cpp

```diff
@@ -24,12 +24,26 @@
 #include "Search_UnitTestHelpers.hpp"
 #include <mpi.h>
 
+namespace ArborX
+{
+
+KOKKOS_FUNCTION constexpr bool operator==(PairIndexRank lhs, PairIndexRank rhs)
+{
+  return lhs.index == rhs.index && lhs.rank == rhs.rank;
+}
+
+KOKKOS_FUNCTION constexpr bool operator<(PairIndexRank lhs, PairIndexRank rhs)
+{
+  return lhs.rank < rhs.rank || (lhs.rank == rhs.rank && lhs.index < rhs.index);
+}
+} // namespace ArborX
```

Comment:
Move it to `ArborXPairIndexRankComparison.hpp` and drop the `KOKKOS_FUNCTION` mark up if it is not needed

Reply:
Why create a separate header when it's used only in one place right now?

Reply:
We agreed to make them private friend functions for now.